apollo-api/src/commonMain/kotlin/com/apollographql/apollo3/api/ApolloRequest.kt

```diff
@@ -73,6 +73,10 @@ private constructor(
       this.enableAutoPersistedQueries = enableAutoPersistedQueries
     }
 
+    override fun canBeBatched(canBeBatched: Boolean?): Builder<D> = apply {
+      if (canBeBatched != null) addHttpHeader(ExecutionOptions.CAN_BE_BATCHED, canBeBatched.toString())
+    }
```

Comment:
Sorry I'm late to the party 🙈:

Should we also add

```kotlin
override val canBeBatched: Boolean
```

I think that would:
1. make it possible to hide CAN_BE_BATCHED as an implementation detail completely
2. keep the symmetry between mutable and immutable `ExecutionOptions`. i.e. if we can mutate it, we should also be able to get it?

Reply:
For 2., good point. But for 1. we still need `CAN_BE_BATCHED` from the Interceptor unfortunately (being an HttpInterceptor it has access to the headers, but not to the `ApolloRequest`)

Reply:
Gotcha 👍 I still think it makes sense to add it but no strong feelings there. Your call.